### PR TITLE
API: Add coverage for content view filter create

### LINKED ISSUE #2462: Automate : content-view filter create in api
please automate:

POST /katello/api/content_view_filters/:content_view_filter_id/rules
Create a filter rule. The parameters included should be based upon the filter type.
Params
Param name 	Description
content_view_filter_id
required 	

filter identifier

Value: string from 2 to 128 characters containting only alphanumeric characters, space, '_', '-' with no leading or trailing space..
name
optional 	

package or package group: name

Value: String
version
optional 	

package: version

Value: String
min_version
optional 	

package: minimum version

Value: String
max_version
optional 	

package: maximum version

Value: String
errata_id
optional 	

erratum: id

Value: String
errata_ids
optional 	

erratum: IDs or a select all object

Value: Must be an array of any type
start_date
optional 	

erratum: start date (YYYY-MM-DD)

Value: String
end_date
optional 	

erratum: end date (YYYY-MM-DD)

Value: String
types
optional 	

erratum: types (enhancement, bugfix, security)

Value: Must be an array of any typ

### PR DESCRIPTION
**Pull request by abalakh**
Added coverage for content view filter create
Closes #2462
Tests results:
```
nosetests tests/foreman/api/test_contentviewfilter.py -m test_create
..............................................
----------------------------------------------------------------------
Ran 46 tests in 369.890s

OK
```

### CODE REVIEW

File: tests/foreman/api/test_contentviewfilter.py [hunk 1]
```diff
@@ -57,3 +81,265 @@ def test_get_with_bad_args(self):
...
         )
+
+    @data(
+        gen_string('alphanumeric', randint(1, 255)),
+        gen_string('alpha', randint(1, 255)),
+        gen_string('cjk', randint(1, 85)),
+        gen_string('latin1', randint(1, 255)),
+        gen_string('numeric', randint(1, 255)),
+        gen_string('utf8', randint(1, 85)),
+        gen_string('html', randint(1, 85)),
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentviewfilter.py]: I see this same bit of data being passed in to every single function. You could clean this up by using a private function. I think this works:

```python
def _data():
    return (
        gen_string('alphanumeric', randint(1, 255)),
        …
    )

@data(*_data())
def test_create_erratum_cvf_with_different_names(self, name):
    …
```

Even better, see whether there's some even more general function that can be used, much as the CLI tests do.

File: tests/foreman/api/test_contentviewfilter.py [hunk 2]
```diff
@@ -57,3 +81,265 @@ def test_get_with_bad_args(self):
...
+        @Assert: Content view filter created successfully and has correct
+        description
+
+        @Feature: Content View Filter - Create
+
+        """
+        cvf = entities.RPMContentViewFilter(
+            content_view=self.content_view,
+            description=description,
+            name=self.cvf_name,
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentviewfilter.py]: Is there any reason to pass in an explicit name here? If a name is not supplied, the `create_missing` function will do so for you.

File: tests/foreman/api/test_contentviewfilter.py [hunk 3]
```diff
@@ -57,3 +81,265 @@ def test_get_with_bad_args(self):
...
+        @Assert: Content view filter created successfully and has repository
+        assigned
+
+        @Feature: Content View Filter - Create
+
+        """
+        cvf = entities.RPMContentViewFilter(
+            content_view=self.content_view,
+            inclusion=True,
+            name=self.cvf_name,
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentviewfilter.py]: Same question about name.

File: tests/foreman/api/test_contentviewfilter.py [hunk 4]
```diff
@@ -57,3 +81,265 @@ def test_get_with_bad_args(self):
...
+        @Assert: Content view filter created successfully and has 'original
+        packages' value
+
+        @Feature: Content View Filter - Create
+
+        """
+        cvf = entities.RPMContentViewFilter(
+            content_view=self.content_view,
+            inclusion=True,
+            name=self.cvf_name,
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentviewfilter.py]: Same question about name.

File: tests/foreman/api/test_contentviewfilter.py [hunk 5]
```diff
@@ -57,3 +81,265 @@ def test_get_with_bad_args(self):
...
+            product=self.product.id,
+            url=DOCKER_REGISTRY_HUB,
+        ).create()
+        self.content_view.set_repository_ids(
+            repo_ids=[self.repo.id, docker_repository.id]
+        )
+        cvf = entities.RPMContentViewFilter(
+            content_view=self.content_view,
+            inclusion=True,
+            name=self.cvf_name,
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentviewfilter.py]: Same question about name.

File: tests/foreman/api/test_contentviewfilter.py [hunk 6]
```diff
@@ -57,3 +81,265 @@ def test_get_with_bad_args(self):
...
+
+        @Assert: Content view filter is not created
+
+        @Feature: Content View Filter - Create
+
+        """
+        with self.assertRaises(HTTPError):
+            entities.RPMContentViewFilter(
+                content_view=self.content_view,
+                name=self.cvf_name,
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentviewfilter.py]: Same question about name.

File: tests/foreman/api/test_contentviewfilter.py [hunk 7]
```diff
@@ -1,20 +1,44 @@
 """Unit tests for the ``content_view_filters`` paths.
 
 A full API reference for content views can be found here:
-http://theforeman.org/api/apidoc/v2/content_view_filters.html
+http://www.katello.org/docs//api/apidoc/content_view_filters.html
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentviewfilter.py]: Replace `docs//` with `docs/`.

**Comment by Ichimonji10** [comment]
ACK pending comments

File: tests/foreman/api/test_contentviewfilter.py [hunk 8]
```diff
@@ -1,20 +1,44 @@
...
 
 """
+from ddt import ddt
```
Comment (elyezer) [inline tests/foreman/api/test_contentviewfilter.py]: Please move `from ... import ...` statements below import statement

File: tests/foreman/api/test_contentviewfilter.py (hunk 1 above)
Comment (elyezer) [inline tests/foreman/api/test_contentviewfilter.py]: :+1:

File: tests/foreman/api/test_contentviewfilter.py (hunk 2 above)
Comment (elyezer) [inline tests/foreman/api/test_contentviewfilter.py]: :+1:

File: tests/foreman/api/test_contentviewfilter.py [hunk 9]
```diff
@@ -57,3 +81,265 @@ def test_get_with_bad_args(self):
...
+
+        @Assert: Content view filter is not created
+
+        @Feature: Content View Filter - Create
+
+        """
+        with self.assertRaises(HTTPError):
+            entities.RPMContentViewFilter(
+                content_view=None,
+                name=self.cvf_name,
```
Comment (elyezer) [inline tests/foreman/api/test_contentviewfilter.py]: The name could be dropped as NailGun will generate one for you.

File: tests/foreman/api/test_contentviewfilter.py [hunk 10]
```diff
@@ -57,3 +81,265 @@ def test_get_with_bad_args(self):
...
+        @Assert: Content view filter created successfully and has correct
+        inclusion value
+
+        @Feature: Content View Filter - Create
+
+        """
+        cvf = entities.RPMContentViewFilter(
+            content_view=self.content_view,
+            inclusion=inclusion,
+            name=self.cvf_name,
```
Comment (elyezer) [inline tests/foreman/api/test_contentviewfilter.py]: NailGun will generate a name for you, this can be dropped.

**Comment by elyezer** [comment]
ACK, pending comments.

**Comment by abalakh** [comment]
Thanks for comments! Updated & rebased. Tests results:
```
nosetests tests/foreman/api/test_contentviewfilter.py -m test_create
..............................................
----------------------------------------------------------------------
Ran 46 tests in 370.193s

OK
```

**Comment by Ichimonji10** [comment]
All comments addressed.

**Pull request merged**